ime/app/src/main/java/com/anysoftkeyboard/gesturetyping/GestureTypingDetector.java

```diff
@@ -318,21 +333,23 @@ public ArrayList<String> getCandidates() {
       }
     }
 
-    if (startKey == null) {
-      Logger.w(TAG, "Could not find a key that is inside %d,%d", corners[0], corners[1]);
-      return mCandidates;
-    }
-
     mCandidateWeights.clear();
     int dictionaryWordsCornersOffset = 0;
     for (int dictIndex = 0; dictIndex < mWords.size(); dictIndex++) {
       final char[][] words = mWords.get(dictIndex);
       final int[] wordFrequencies = mWordFrequencies.get(dictIndex);
       for (int i = 0; i < words.length; i++) {
-        // Check if current word would start with the same key
+        // Check if current word starts with a key close to the gesture start point
         final Keyboard.Key wordStartKey = mKeysByCharacter.get(Dictionary.toLowerCase(words[i][0]));
-        // filtering all words that do not start with the initial pressed key
-        if (wordStartKey != startKey) {
+        if (wordStartKey == null) {
+          continue; // Character not found on keyboard
+        }
+
+        // Calculate squared distance from gesture start to word's starting key
+        final int distanceSquared = wordStartKey.squaredDistanceFrom(corners[0], corners[1]);
```

Comment:
Micro-optimization:
first check if the `wordStartKey != startKey` and if so then calculate the distance and do the other stuff.

Reply:
If you are not worried about this, or do not agree, lmk